plugins/builds/index.js

```diff
@@ -720,17 +669,39 @@ async function createOrRunNextBuild({ buildFactory, jobFactory, eventFactory, pi
     };
 
     /* CHECK WHETHER NEXT BUILD EXISTS */
-    const nextBuild = await getNextBuild({
-        isExternal,
-        pipelineFactory,
-        externalPipelineId,
-        externalJobName,
-        jobFactory,
-        eventFactory,
-        event,
-        workflowGraph,
-        nextJobName
-    });
+    let nextBuild;
+
+    // If next build is external, return the latest build with same job ID
+    if (isExternal) {
+        const p = await pipelineFactory.get(externalPipelineId);
+        const jobArray = await p.getJobs({ params: { name: externalJobName } });
+        const j = await jobFactory.get(jobArray[0].id);
+
+        nextBuild = await j.getLatestBuild({ status: 'CREATED' });
```

Comment:
will this be accurate ? shouldn't it lookup based on the original event  ID  in `externalPipeline`?

ie rather than doing `job/builds` we should grab `nextBuild` from `event/builds`

Reply:
updated